crates/ast/src/lang/core/def/def.rs

```diff
@@ -574,7 +574,7 @@ fn canonicalize_pending_def<'a>(
                                     ..
                                 } => {
                                     if arguments.len() != type_arguments.len() {
-                                        panic!("argument number mismatch");
+                                        internal_error!("argument number mismatch");
```

Comment:
Same with this and the errors below in this file. They might really be `user_error!`